### PR TITLE
Deploy gh-pages branch from travis

### PR DESCRIPTION
**Pull request by mnapoli**
### WIP for #34 

(this is #44 but merged into a local branch so that we can all push commits to it and we can test the results in this repository)

To safely deploy with Couscous from a Travis build we need to take this things into consideration:

- [x] Make sure the deployment is running only for master.
- [x] Make sure it is not a pull request.
- [x] Make sure we deploy once per build. E.G. don't deploy for every PHP version tested :smile:
- [x] Decide what to use to deploy: [Private Key](http://stackoverflow.com/questions/23277391/pushing-to-github-from-travis-ci) or [Personal Access Token](http://stackoverflow.com/questions/18027115/committing-via-travis-ci-failing)  
    - More info on [Personal Access Token](https://medium.com/@nthgergo/publishing-gh-pages-with-travis-ci-53a8270e87db)
- [x] Write documentation in `docs/travis.md`
- [ ] ~~Couscous specific problem: don't generate & upload `couscous.phar` for pushes to master (else we might push an unstable phar)~~ (will be done in a new PR)

The process is detailed in [docs/travis.md](https://github.com/CouscousPHP/Couscous/blob/travis/docs/travis.md)

Feel free to add commits and keep working on this together.

### CODE REVIEW

**Comment by wysow** [comment]
@CouscousPHP/collaborators This should be ready for full review and hopefully merge... If needed I can squash some commits...

File: couscous.yml [hunk 1]
```diff
@@ -8,11 +8,10 @@ exclude:
 
 scripts:
     before:
-        - bin/compile
-        - cp bin/couscous.phar website/
         - lessc --clean-css website/less/main.less website/css/all.min.css
     after:
         - rm website/couscous.phar
+        - bin/couscous deploy-binaries
```
Comment (mnapoli) [inline couscous.yml]: @wysow did this line was left by mistake?

**Comment by mnapoli** [comment]
Except the inline comment LGTM :+1:!

File: couscous.yml (hunk 1 above)
Comment (wysow) [inline couscous.yml]: @mnapoli Sure sorry for this, deleted now ;)

**Comment by wysow** [comment]
@CouscousPHP/collaborators Feel free to review and merge if it's ok for you guys! ;)

**Comment by mnapoli** [comment]
Let's ship it!

     .  o ..                  
     o . o o.o                
          ...oo               
            __[]__            
         __|_o_o_o\__         
         \""""""""""/         
          \. ..  . /          
     ^^^^^^^^^^^^^^^^^^^^

**Pull request merged**